### PR TITLE
Add instructions to install NixOS

### PR DESCRIPTION
**Pull request by kekrby**
Hi everybody!

Firstly, I want to than you all for maintaining such a great project!

I wanted to run NixOS on my MacBook Pro, but unfortunately the existing solution was outdated and I wanted to get familiar with Nix so I decided to build a working solution from scratch. I am happy to announce that I managed to get NixOS to work quite well. Audio, bluetooth, WiFi and pretty much all the things you would expect from a computer work well (except suspend).

My changes are available in [my fork of `nixos-hardware`](https://github.com/kekrby/nixos-hardware) and you can use [this repo](https://github.com/kekrby/nixos-t2-iso) to build an installer.

I am planning to upstream my changes once they are tested by others. It would be great if one of you could mention it in the discord server so that more people are aware of this and can test it. (I was going to ask myself but after creating an account it got falsely flagged as spam and now I am waiting for discord support to fix it.)

Best regards,
kekrby

### CODE REVIEW

File: docs/distributions/nixos/installation.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+5. Partition your disk using `gdisk`, initialize the partitions with the `mkfs` command of the filesystem you want (`mkswap` is for swap) and mount them under `/mnt`.
+    * **Note**: You might want to leave a little part of your disk as a FAT32 partition to be able to transfer files easily between MacOS and Linux.
+6. Connect to internet using `nmcli`.
+7. Generate your configuration using `sudo nixos-generate-config --root /mnt`.
+8. Edit `/mnt/etc/configuration.nix` to your liking. Add `"${builtins.fetchGit { url = "https://github.com/kekrby/nixos-hardware.git"; }}/apple/t2"` to `imports`.
+    * **Note**: Don't forget to add a bootloader, `systemd-boot` works quite well. If you want to use `GRUB`, don't forget to set `boot.grub.efiInstallAsRemovable`, `boot.grub.efiSupport` to `true` and `boot.grub.device` to `"nodev"`.
+9. `cd` into `/mnt` and run `sudo nixos-install`.
+
+And the installation is complete!
+Note that you should probably transition to a more structured configuration using flakes.
```
Comment (Redecorating) [inline docs/distributions/nixos/installation.md:22]: Are there instructions on how to do this?

**Comment by Redecorating** [comment]
https://github.com/kekrby/nixos-hardware/blob/761380586df33b1d2f31c56784c1ce9aff767afe/apple/t2/default.nix#L12 here, users with MacbookPro16,4 also need the same audio files as MacbookPro16,1

**Comment by EmreTech** [comment]
#29 should be closed if this PR were to be merged

File: docs/distributions/nixos/installation.md (hunk 1 above)
Comment (kekrby) [reply docs/distributions/nixos/installation.md:22]: I think [this](https://github.com/NixOS/nixos-hardware/blob/master/README.md#using-nix-flakes-support) sums it up well. I'll add a link to it.

**Comment by kekrby** [comment]
I think this is ready to merge, the guide looks pretty complete. I squashed all my changes into one commit.

**Comment by Redecorating** [comment]
Sorry for the delay, I was hoping that someone would test it, but if it works for you it should be fine. I've requested a review from the person who was previously maintaining nix for t2 macs just in case there are any issues he notices (I don't know nix that well).

Also, you said discord wasn't working for you, would Matrix or IRC work better for you? (Bridging discord to those is a possibility)

File: docs/distributions/nixos/installation.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+* A USB that can be plugged in your Mac.
+
+## Installation Steps
+
+1. Get a NixOS iso from the releases page of [this repo](https://github.com/kekrby/nixos-t2-iso) and write it to your USB using `dd` or another tool.
+2. Partition you disk using Disk Utily by splitting the main APFS partition into two partitions where one of them is a FAT32 partition with the amount of space you want to allocate for your Linux installation.
+3. [Open the Startup Security Utility](https://support.apple.com/en-us/HT208198) and set the security level to "no security".
+4. Reboot while pressing the `Option` key and select the orange `EFI Boot` option.
+5. Partition your disk using `gdisk`, initialize the partitions with the `mkfs` command of the filesystem you want (`mkswap` is for swap) and mount them under `/mnt`.
```
Comment (mikroskeem) [inline docs/distributions/nixos/installation.md:13]: I'd rather suggest `cfdisk` to beginners, as it's easier & more straightforward to use. It's also available on NixOS installer out of the box.

File: docs/distributions/nixos/installation.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    * **Note**: You might want to leave a little part of your disk as a FAT32 partition to be able to transfer files easily between MacOS and Linux.
+6. Connect to internet using `nmcli`.
+7. Generate your configuration using `sudo nixos-generate-config --root /mnt`.
+8. Edit `/mnt/etc/configuration.nix` to your liking. Add `"${builtins.fetchGit { url = "https://github.com/kekrby/nixos-hardware.git"; }}/apple/t2"` to `imports`.
+    * **Note**: Don't forget to add a bootloader, `systemd-boot` works quite well. If you want to use `GRUB`, don't forget to set `boot.grub.efiInstallAsRemovable`, `boot.grub.efiSupport` to `true` and `boot.grub.device` to `"nodev"`.
+9. `cd` into `/mnt` and run `sudo nixos-install`.
+
+And the installation is complete!
+Note that you should probably transition to a more structured configuration [using flakes](https://github.com/NixOS/nixos-hardware/blob/master/README.md#using-nix-flakes-support), that is ommited here for brevity.
```
Comment (mikroskeem) [inline docs/distributions/nixos/installation.md:22]:
```suggestion
Note that you should probably transition to a more structured configuration [using flakes](https://github.com/NixOS/nixos-hardware/blob/master/README.md#using-nix-flakes-support), that is omitted here for brevity.
```

File: docs/distributions/nixos/installation.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+3. [Open the Startup Security Utility](https://support.apple.com/en-us/HT208198) and set the security level to "no security".
+4. Reboot while pressing the `Option` key and select the orange `EFI Boot` option.
+5. Partition your disk using `gdisk`, initialize the partitions with the `mkfs` command of the filesystem you want (`mkswap` is for swap) and mount them under `/mnt`.
+    * **Note**: You might want to leave a little part of your disk as a FAT32 partition to be able to transfer files easily between MacOS and Linux.
+6. Connect to internet using `nmcli`.
+7. Generate your configuration using `sudo nixos-generate-config --root /mnt`.
+8. Edit `/mnt/etc/configuration.nix` to your liking. Add `"${builtins.fetchGit { url = "https://github.com/kekrby/nixos-hardware.git"; }}/apple/t2"` to `imports`.
+    * **Note**: Don't forget to add a bootloader, `systemd-boot` works quite well. If you want to use `GRUB`, don't forget to set `boot.grub.efiInstallAsRemovable`, `boot.grub.efiSupport` to `true` and `boot.grub.device` to `"nodev"`.
+9. `cd` into `/mnt` and run `sudo nixos-install`.
```
Comment (mikroskeem) [inline docs/distributions/nixos/installation.md:19]:
```suggestion
9. Now run `sudo nixos-install`.
```

`nixos-install` operates on `/mnt` (see `--root` flag) by default, no need to change directory.

File: docs/distributions/nixos/installation.md [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+## Installation Steps
+
+1. Get a NixOS iso from the releases page of [this repo](https://github.com/kekrby/nixos-t2-iso) and write it to your USB using `dd` or another tool.
+2. Partition you disk using Disk Utily by splitting the main APFS partition into two partitions where one of them is a FAT32 partition with the amount of space you want to allocate for your Linux installation.
+3. [Open the Startup Security Utility](https://support.apple.com/en-us/HT208198) and set the security level to "no security".
+4. Reboot while pressing the `Option` key and select the orange `EFI Boot` option.
+5. Partition your disk using `gdisk`, initialize the partitions with the `mkfs` command of the filesystem you want (`mkswap` is for swap) and mount them under `/mnt`.
+    * **Note**: You might want to leave a little part of your disk as a FAT32 partition to be able to transfer files easily between MacOS and Linux.
+6. Connect to internet using `nmcli`.
```
Comment (mikroskeem) [inline docs/distributions/nixos/installation.md:15]: Looking at the ISO, you're pulling in NetworkManager but it seems to be configured to use `wpa_supplicant`. Isn't `iwd` still more reliable to use?

**Review by mikroskeem — CHANGES_REQUESTED** [review]

**Comment by kekrby** [comment]
No problem at all! 

By the way, yesterday I added prebuilt isos to [this repo](https://github.com/kekrby/nixos-t2-iso) to make testing easier. I think people would be more likely to test it if they were aware of the existence of prebuilt isos as they take quite some time to build. I hope you can announce it in the discord server.

Regarding communications, I don't really have a problem using discord but they marked my account as spam as soon as I created it and they haven't fixed it yet.

File: docs/distributions/nixos/installation.md (hunk 2 above)
Comment (kekrby) [reply docs/distributions/nixos/installation.md:13]: I think `gdisk` is available by default too but you're right, `cfdisk` is more friendly.

**Comment by kekrby** [comment]
Thanks for the review @mikroskeem, I'll make changes accordingly.

**Comment by kekrby** [comment]
I made the changes and squashed them. If @mikroskeem gives the thumbs up, I think it can be merged.

**Review by mikroskeem — APPROVED** [review]
LGTM for now

**Pull request merged**